site/src/pages/UserSettingsPage/SecurityPage/SecurityPage.tsx

```diff
@@ -1,29 +1,21 @@
-import { useMachine } from "@xstate/react";
 import { useMe } from "hooks/useMe";
 import { ComponentProps, FC } from "react";
-import { userSecuritySettingsMachine } from "xServices/userSecuritySettings/userSecuritySettingsXService";
-import { Section } from "../../../components/SettingsLayout/Section";
+import { Section } from "components/SettingsLayout/Section";
 import { SecurityForm } from "./SettingsSecurityForm";
-import { useQuery } from "@tanstack/react-query";
+import { useMutation, useQuery } from "@tanstack/react-query";
 import { getAuthMethods, getUserLoginType } from "api/api";
 import {
   SingleSignOnSection,
   useSingleSignOnSection,
 } from "./SingleSignOnSection";
 import { Loader } from "components/Loader/Loader";
 import { Stack } from "components/Stack/Stack";
+import { updatePassword } from "api/queries/users";
+import { displaySuccess } from "components/GlobalSnackbar/utils";
 
 export const SecurityPage: FC = () => {
   const me = useMe();
-  const [securityState, securitySend] = useMachine(
-    userSecuritySettingsMachine,
-    {
-      context: {
-        userId: me.id,
-      },
-    },
-  );
-  const { error } = securityState.context;
+  const updatePasswordMutation = useMutation(updatePassword());
```

Comment:
Does this need to invalidate any queries in the query client, or is the rest of the querying logic set up to update things frequently enough that it's not much of an issue?

Reply:
No we don't because we don't store the password anywhere.

Reply:
Okay, cool. That was my only concern. I'm going to approve this.